### PR TITLE
rpc: use SOCKS5 proxy if configured

### LINKED ISSUE #1287: RPC connection to dcrd via Tor Proxy not working
I've set up a Tor hidden service on my dcrd node, and when I attempt to connect to it using the --proxy tag in dcrwallet, with the correct onion address and port, the connection is failing:

https://gist.githubusercontent.com/fridaynext/c40a5fc47c44424ebaaa11e865359b4d/raw/c093441a14f9dc584e8bfc949df6a768fc1839ac/dcrwallet.log

It appears dcrwallet is not using the proxy at port 9150 (for Browser, otherwise 9050 for standalone) to attempt to find the .onion address.

### PR DESCRIPTION
**Pull request by aerth**
There are config options and flags that currently exist for using SOCKS proxy, but don't do anything.
This exports `NewRPCClientConfig`, which creates an RPC Client to the `chain` package and uses it in dcrwallet command.

  * `NewRPCClientConfig` accepts a `dcrrpcclient.ConnConfig` for even more precise configuration

Resolves #1287

### CODE REVIEW

File: chain/chain.go [hunk 1]
```diff
@@ -44,26 +44,34 @@ type RPCClient struct {
 // but must be done using the Start method.  If the remote server does not
 // operate on the same decred network as described by the passed chain
 // parameters, the connection will be disconnected.
-func NewRPCClient(chainParams *chaincfg.Params, connect, user, pass string, certs []byte,
-	disableTLS bool) (*RPCClient, error) {
+func NewRPCClient(chainParams *chaincfg.Params, connect, user, pass string,
```
Comment (jrick) [inline chain/chain.go]: This function signature may not be modified in this module's major version.  You will have to find a different, backwards-compatible way of adding this, or make a v2 module.

File: netparams/params.go [hunk 2]
```diff
@@ -14,6 +14,9 @@ type Params struct {
 	JSONRPCClientPort string
 	JSONRPCServerPort string
 	GRPCServerPort    string
+	Proxy             string
+	ProxyUser         string
+	ProxyPass         string
```
Comment (jrick) [inline netparams/params.go]: None of these belong in this package.

File: chain/chain.go (hunk 1 above)
Comment (aerth) [reply chain/chain.go]: fixed

File: chain/chain.go [hunk 3]
```diff
@@ -38,26 +38,47 @@ type RPCClient struct {
...
+}
+
+// NewClient creates a client connection to the server described by the
+// connect string.  If disableTLS is false, the remote RPC certificate must be
+// provided in the certs slice.  The connection is not established immediately,
+// but must be done using the Start method.  If the remote server does not
+// operate on the same decred network as described by the passed chain
+// parameters, the connection will be disconnected. If proxypath is not empty,
+// it will be used to connect.
+func NewClient(chainParams *chaincfg.Params, connect, user, pass string, certs []byte,
```
Comment (jrick) [inline chain/chain.go]: If NewClientConfig is exported, I don't see any reason to also include this.

File: chain/chain.go [hunk 4]
```diff
@@ -38,26 +38,47 @@ type RPCClient struct {
...
+		DisableTLS:           disableTLS,
+		Proxy:                proxypath,
+		ProxyUser:            proxyuser,
+		ProxyPass:            proxypass,
+	})
+}
 
+// NewClientConfig creates a client connection to the server described by the
+// passed chainParams and connConfig
+func NewClientConfig(chainParams *chaincfg.Params, connConfig *dcrrpcclient.ConnConfig) (*RPCClient, error) {
```
Comment (jrick) [inline chain/chain.go]: nitpick: can you rename this to NewRPCClientConfig for more similarity with the existing NewRPCClient?

File: chain/chain.go (hunk 3 above)
Comment (aerth) [reply chain/chain.go]: One reason is that any caller of NewRPCClientConfig will need to import dcrrpcclient package, and any user of NewRPCClient can simply update to using SOCKS without changing too much code.

Comment (jrick) [reply chain/chain.go]: It's a transient dependency anyways, making it an explicit import doesn't change anything and adds less cruft to the public API.

**Pull request merged**